lib/auth/accesspoint/accesspoint.go

```diff
@@ -0,0 +1,155 @@
+/*
+Copyright 2015-2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package accesspoint provides helpers for configuring caches in the context of
+// setting up service-level auth access points. this logic has been moved out of
+// lib/service in order to facilitate better testing practices.
+package accesspoint
+
+import (
+	"context"
+	"slices"
+	"time"
+
+	"github.com/gravitational/trace"
+	log "github.com/sirupsen/logrus"
+	oteltrace "go.opentelemetry.io/otel/trace"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	"github.com/gravitational/teleport/lib/cache"
+	"github.com/gravitational/teleport/lib/observability/tracing"
+	"github.com/gravitational/teleport/lib/services"
+)
+
+// AccessCacheConfig holds parameters used to confiure a cache to
+// serve as an auth access point for a teleport service.
+type AccessCacheConfig struct {
+	// Context is the base context used to propagate closure to
+	// cache components.
+	Context context.Context
+	// Services is a collection of upstream services from which
+	// the access cache will derive its state.
+	Services services.Services
+	// Setup is a function that takes cache configuration and
+	// modifies it to support a specific teleport service.
+	Setup cache.SetupConfigFn
+	// CacheName identifies the cache in logs.
+	CacheName []string
+	// Events is true if cache should turn on events.
+	Events bool
```

Comment:
What does `turn on` mean in this context? "Emit events", maybe?

Sorry, I know this is just copying code around but it was not clear, at least for me.